spec/controllers/shoutouts_controller_spec.rb

```diff
@@ -0,0 +1,245 @@
+                                              time_period_id: time_period.id,
+                                              rich_text: nil,
+                                              recipients: [user2.id, user3.id, user4.id] } }
+        end.not_to change(Shoutout, :count)
+        expect(ShoutoutRecipient.all.length).to be(0)
+      end
+    end
+
+    it 'renders json answer with one recipient' do
+      allow(controller).to receive(:current_user).and_return(user)
+      post :create, params: { shoutout: { user_id: user.id,
+                                          time_period_id: time_period.id,
+                                          rich_text: '<br>',
+                                          recipients: [user5.id] } }
+      expect(ShoutoutRecipient.all.length).to be(1)
+      expect(response.body).to include_json({
+                                              user_id: user.id,
+                                              time_period_id: time_period.id,
+                                              rich_text: '<br>',
+                                              recipients: a_collection_including(user5.id.to_s)
+                                            })
+    end
+
+    it 'renders json answer with some recipients' do
+      allow(controller).to receive(:current_user).and_return(user)
+      post :create, params: { shoutout: { user_id: user.id,
+                                          time_period_id: time_period.id,
+                                          rich_text: '<br>',
+                                          recipients: [user2.id, user3.id, user4.id] } }
+      expect(ShoutoutRecipient.all.length).to be(3)
+      expect(response.body).to include_json({
+                                              user_id: user.id,
+                                              time_period_id: time_period.id,
+                                              rich_text: '<br>',
+                                              recipients: a_collection_including(user2.id.to_s, user3.id.to_s, user4.id.to_s)
+                                            })
+    end
+  end
+  describe 'PATCH #update' do
+    let!(:user) { create :user, opt_out: false }
+    let!(:user1) { create :user, opt_out: false }
+    let!(:time_period) { create :time_period }
+    let!(:time_period1) { create :time_period }
+    let!(:shoutout) do
+      create(:shoutout, { id: 1,
+                          user_id: user.id,
+                          time_period_id: time_period.id,
+                          rich_text: '<br>',
+                          recipients: [user2.id.to_s, user3.id.to_s, user4.id.to_s],
+                          digest: 1234567890 })
+    end
+    let!(:recipients) do
+      create(:shoutout_recipient, { shoutout_id: 1, user_id: user2[:id] })
+      create(:shoutout_recipient, { shoutout_id: 1, user_id: user3[:id] })
+      create(:shoutout_recipient, { shoutout_id: 1, user_id: user4[:id] })
+    end
```

Comment:
please write `user2.id` instead of `user2[:id]`

Reply:
fixed